src/chapters/angular-lsn2/ngFor.rst

```diff
@@ -0,0 +1,206 @@
+
+Just like a ``for`` loop in JavaScript requires a specific syntax in order to
+operate, loops in Angular must follow a set of rules. Let's explore these rules
+by adding ``ngFor`` to our movie list code.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <div class='movies'>
+      <h3>Movies to Watch</h3>
+      <ol>
+         <li *ngFor = "let movie of movies">{{movie}}</li>
+      </ol>
+   </div>
+
+Some items to note:
+
+#. Structural directives all begin with the ``*`` symbol.
+#. The string ``"let movie of movies"`` provides the instructions
+   for running the loop.
+
+   a. The ``let`` keyword declares the ``movie`` variable.
+   b. ``of movies`` sets ``movie`` equal to the first element of the ``movies``
+      array. Each iteration of the loop sets ``movie`` equal to the next title
+      in the array.
+
+#. The ``*ngFor`` statement is placed INSIDE the ``<li>`` tag.
+#. ``{{movie}}`` is the placeholder for the current value of ``movie``.
+
+By placing the ``*ngFor`` statement inside the tag, the loop generates
+multiple ``<li></li>`` elements. Each iteration adds a new list item to the
+HTML code, one for each title in the ``movies`` array.
+
+.. admonition:: Warning
+
+   The ``*ngFor`` statement generates a new HTML tag for each item in the
+   array. *Be careful where you put the statement!* If we had added
+   ``*ngFor = "let movie of movies"`` to the ``<h3>`` tag, then the ``Movies
+   To Watch`` title would have been repeated multiple times.
+
+In general, the syntax for ``*ngFor`` is:
+
+::
+
+   *ngFor = "let variableName of arrayName"
+
+Where ``variableName`` is the loop variable, and ``arrayName`` represents the
+array to iterate through.
+
+``*ngFor`` only operates over the contents of an array. If we want to iterate
+over the characters in a string or the key/value pairs in an object, then we
+must first convert them into arrays.
+
+Try It
+-------
+
+From the ``lesson2`` folder in VSCode, open the
+``examples/ngfor-practice/src/app/chores`` folders and select the
+``chores.component.html`` file.
+
+.. figure:: ./figures/ngfor-menu.png
+   :alt: Access ngFor practice in VSCode.
+
+The starter code should match this:
+
+.. sourcecode:: html
+   :linenos:
+
+   <div class='chores'>
+      <h3>Chores To Do Today</h3>
+      <ul>
+         <li>{{chores[0]}}</li>
+         <li>{{chores[1]}}</li>
+         <li>{{chores[2]}}</li>
+      </ul>
+      <hr>
+   </div>
+
+In the VSCode terminal window, navigate to the ``ngfor-practice`` folder.
+
+.. sourcecode:: bash
+
+   $ pwd
+      angular-lc101-projects/lesson2
+   $ ls
+      examples        exercises       studio
+   $ cd examples
+   $ ls
+      ngfor-practice  ngif-practice
+   $ cd ngfor-practice
+
+Enter ``ng serve`` to launch the project, then:
+
+#. Modify ``chores.component.html`` with ``*ngFor`` to loop over the items
+   stored inside the ``chores`` array.
+#. Open ``chores.component.ts``. Add or remove items to the ``chores``
```

Comment:
3. I think the chores list should be an array of objects. Check out my suggested solution in my FORK https://github.com/welzie/angular-lc101-projects/pull/1

Reply:
This is similar to my thinking for the bonus task, only I used a 2-D array of chores plus a separate array for the titles.  The object is the better way to go for the task, however.

Still, for the non-bonus step I think the "array of chores and array of titles" approach is simpler as a starting point.  Let's put our heads together on the options.

Reply:
As discussed let's keep it using two arrays, but change the bonus to use an array of objects.

Reply:
Added the object idea to the bonus task.